static/app/views/issueList/customViewsHeader.tsx

```diff
@@ -241,16 +241,8 @@ function CustomViewsIssueListHeaderTabsContent({
       tabListState?.setSelectedKey('temporary-tab');
       return;
     }
-  }, [
-    tabListState,
-    draggableTabs,
-    navigate,
-    organization.slug,
-    query,
-    queryParams,
-    sort,
-    viewId,
-  ]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [tabListState, navigate, organization.slug, query, sort, viewId]);
```

Comment:
Could this still cause the same issue since were calling `tabListState.setSelectedKey()` inside which will cause `tabListState` to update and refire the hook?

Reply:
Good point - intuitively I feel like it should too, but in my testing that doesn't occur. I'm pretty sure this is because this `useEffect`'s behavior depends on the url query, and if `tabListState.setSelectedKey()` is called, it changes the query in a way such that the subsequent time this useEffect runs, it would not result in `tabListState.setSelectedKey()` being called again. Sorta like reaching a state of equilibrium after one unnecessary rerender if that makes sense?

I think you've still got a point that leaving tabListState as a dependency is more likely to cause issues down the line, even if its stable now, so I'll remove it in the next commit.